lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsOutputFile.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.gcs;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.Storage.BlobTargetOption;
+import com.google.common.collect.ImmutableList;
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoOutputFile;
+import io.trino.memory.context.AggregatedMemoryContext;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.FileAlreadyExistsException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.filesystem.gcs.GcsUtils.getBlob;
+import static io.trino.filesystem.gcs.GcsUtils.handleGcsException;
+import static java.util.Objects.requireNonNull;
+
+public class GcsOutputFile
+        implements TrinoOutputFile
+{
+    private final GcsLocation location;
+    private final Storage storage;
+    private final long writeBlockSizeBytes;
+
+    public GcsOutputFile(GcsLocation location, Storage storage, long writeBlockSizeBytes)
+    {
+        this.location = requireNonNull(location, "location is null");
+        this.storage = requireNonNull(storage, "storage is null");
+        checkArgument(writeBlockSizeBytes >= 0, "writeBlockSizeBytes is negative");
+        this.writeBlockSizeBytes = writeBlockSizeBytes;
+    }
+
+    @Override
+    public OutputStream create(AggregatedMemoryContext memoryContext)
+            throws IOException
+    {
+        return createOutputStream(memoryContext, false);
+    }
+
+    @Override
+    public OutputStream createOrOverwrite(AggregatedMemoryContext memoryContext)
+            throws IOException
+    {
+        return createOutputStream(memoryContext, true);
+    }
+
+    @Override
+    public OutputStream createExclusive(AggregatedMemoryContext memoryContext)
+            throws IOException
+    {
+        return create(memoryContext);
+    }
+
+    private OutputStream createOutputStream(AggregatedMemoryContext memoryContext, boolean overwrite)
+            throws IOException
+    {
+        try {
+            ImmutableList.Builder<BlobTargetOption> optionsBuilder = ImmutableList.builder();
```

Comment:
You can use `ArrayList`. No need to build an immutable list just to turn it into an array.

Reply:
Changed to use static variables (above) - lmk if that works.